docs/book/src/SUMMARY.md

```diff
@@ -0,0 +1,12 @@
+# Summary
```

Comment:
We can later explore making summary autogenerated, as this may be outdated with manual edits in the future.

Reply:
Sounds good. When we work on publishing this as a book, we'll have to add build scripts (upstream uses [Makefile](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/docs/book/Makefile)) and we can also work on having an automatically updated summary.